app/lib/session_encryptor.rb

```diff
@@ -0,0 +1,64 @@
+class SessionEncryptor
+  KEY_LEN = ActiveSupport::MessageEncryptor.key_len
```

Comment:
this is interesting, so what's the MessageEncryptor here? It's a built in from active support?

Does that mean it's a standard length or does ActiveSupport read this out of the config file?

Would be interested to know what's going on here?

Reply:
Each cipher has a standard length (https://apidock.com/rails/ActiveSupport/MessageEncryptor/key_len/class), the default cipher is AES-256-CBC (https://github.com/rails/rails/blob/2bbb519482dbc7969321719cb515216cf1515bfc/activesupport/lib/active_support/message_encryptor.rb#L87)

Reply:
The crypto here is more or less taken straight from the example in https://apidock.com/rails/ActiveSupport/MessageEncryptor

The only new bits are:

1. The JSON encoding / decoding
2. Storing the salt in the returned value